### ISSUE TITLE
Nested folders are not deleted in OneDrive delta incremental backups

### ISSUE THREAD

**Issue by meain** [alcionai/corso#2558]
Deleting nested folders does not delete it from the kopia backup snapshot, and any followup operation within the parent results in the following error:

```
Error: 1 error occurred:
        * Failed to run OneDrive backup for user Jo...aL@8...rj.onmicrosoft.com: doing backup: merging details: incomplete migration of backup details
```

The item within the folder seems to get deleted from the details file as well the folder is deleted from paths in the metadata, but the actual folder and the contents within it still remains in the backup snapshot.

##### Repro

Create the following folder structure:

```
folder1
- folder2
  - file
```

Do a full backup with this, delete `folder2` and do an incremental backup. `folder2` still remains in the snapshot.

##### Observations

A collection is created and passed to kopia with the correct state.

**Comment by ashmrtn** [comment]
hmmm could this have been an intermediate state that the code was in? I've taken the branch in #2604 (commit e3488b9a5c1f952f2f743306800d62f2d669c2bc), made a new branch, cherry-picked #2496 and #2497, and updated the backup version so #2496 works properly. I've pushed the code I ran to a separate branch and used commit 62d83d2df05feba4637f0fb3cfc1b9c535414727 from it to test with. This new branch didn't exhibit the behavior you saw

The folder hierarchy was
```text
- root
  - folder1
    - folder11
      - file.txt
```

Both backups were run with `--enable-onedrive-incrementals`. Logs show the first backup attempted hierarchy merging but had no base to merge with (expected) while the second backup did use a base snapshot to merge the hierarchy
